src/coreclr/src/jit/flowgraph.cpp

```diff
@@ -4016,9 +4016,6 @@ BasicBlock* Compiler::fgCreateGCPoll(GCPollType pollType, BasicBlock* block)
             value = gtNewIndOfIconHandleNode(TYP_INT, (size_t)addrTrap, GTF_ICON_GLOBAL_PTR, false);
         }
 
-        // Treat the reading of g_TrapReturningThreads as volatile.
-        value->gtFlags |= GTF_IND_VOLATILE;
```

Comment:
in c++ an equivalent load is done via `LoadWithoutBarrier()` to ensure that the
program order of the load is preserved. (not hoisted out of a loop or cached in a local, for example)

So we only need to ensure that this load is not optimizable. 
I have confirmed with the JIT team that it is not optimizable for several reasons:
- we introduce the load really late in terms of JIT stages - after all major optimizations are done 
- the load feeds a branch and the location is formally unknown

since noone could optimize the load, no special flags are needed.

I've added a comment explaining this instead.